### PR TITLE
Introduce built-in number operations in Qute extension

### LINKED ISSUE #7283: qute: modulo operation in iterations
**Description**
`
{#for printable in printables}
<div>{printable.name}</div>   
{#if count % 5 == 0 }
           <div class="clearfix"></div>
   {/if}
{/for}
`

### PR DESCRIPTION
**Pull request by gastaldi**
Fixes #7283

### CODE REVIEW

**Comment by gsmet** [comment]
I let Martin validate the approach but if we go with this, we need a documentation update.

**Comment by gastaldi** [comment]
I updated the documentation to reflect the changes

**Comment by gastaldi** [comment]
Seems that CI is not being triggered. Let me close and reopen the PR just in case

**Comment by gsmet** [comment]
@mkouba can you check that one, see if that's what we want to do?

**Comment by mkouba** [comment]
@gsmet I'm pretty sure I added some comments yesterday but they're gone :-(. I'm going to add them again.

File: extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/extensions/CollectionTemplateExtensions.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import io.quarkus.qute.TemplateExtension;
 
 @TemplateExtension
-public class BuiltinTemplateExtensions {
+public class CollectionTemplateExtensions {
```
Comment (mkouba) [inline extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/extensions/CollectionTemplateExtensions.java]: `Map` is not a collection so we should probably extract the relevant methods in `MapTemplateExtensions`.

File: extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/extensions/IntegerTemplateExtensions.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package io.quarkus.qute.runtime.extensions;
+
+import io.quarkus.qute.TemplateExtension;
+
+@TemplateExtension
+public class IntegerTemplateExtensions {
```
Comment (mkouba) [inline extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/extensions/IntegerTemplateExtensions.java]: Maybe we should use a more generic name, ie. `NumberTemplateExtensions`?

**Review by mkouba — CHANGES_REQUESTED** [review]
The approach is probably OK. At least for now. We should re-evaluate the builtins again once the API is stabilized enough.

File: docs/src/main/asciidoc/qute.adoc [hunk 3]
```diff
@@ -69,13 +71,16 @@ public class HelloResource {
     }
 }
 ----
-<1> If there is no `@ResourcePath` qualifier provided, the field name is used to locate the template. In this particular case, we're injecting a template with path `templates/hello.txt`.
-<2> `Template.data()` returns a new template instance that can be customized before the actual rendering is triggered. In this case, we put the name value under the key `name`. The data map is accessible during rendering. 
+<1> If there is no `@ResourcePath` qualifier provided, the field name is used to locate the template.
+In this particular case, we're injecting a template with path `templates/hello.txt`.
```
Comment (mkouba) [inline docs/src/main/asciidoc/qute.adoc]: Is it legal to use line breaks in source code callouts? I thought that you always need to use `+` here...

Comment (gastaldi) [reply docs/src/main/asciidoc/qute.adoc]: Hm, not sure why this part was changed, maybe IntelliJ did it when I saved the doc

File: docs/src/main/asciidoc/qute.adoc [hunk 4]
```diff
@@ -171,6 +171,37 @@ public class ItemResource {
...
 <3> A static template extension method can be used to add "computed properties" to a data class. The class of the first parameter is used to match the base object and the method name is used to match the property name.
 
+== Built-in Template Extension methods
+
+Quarkus provides a set of built-in extension methods.
+
+=== Map extension methods:
+
+* `keys()` or `keySet()`: Returns a Set view of the keys contained in a map
+** `{#for key in map.keySet()}`
```
Comment (mkouba) [inline docs/src/main/asciidoc/qute.adoc]: The brackets are optional and I'd say that `map.keySet` is more readable. The same applies to all snippets.

File: docs/src/main/asciidoc/qute.adoc [hunk 5]
```diff
@@ -171,6 +171,37 @@ public class ItemResource {
 <2> Make the `Item` object accessible in the template.
 <3> A static template extension method can be used to add "computed properties" to a data class. The class of the first parameter is used to match the base object and the method name is used to match the property name.
 
+== Built-in Template Extension methods
```
Comment (mkouba) [inline docs/src/main/asciidoc/qute.adoc]: I think that we should move this section to the reference guide instead: https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/qute-reference.adoc

**Review by mkouba — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/qute.adoc (hunk 5 above)
Comment (gastaldi) [reply docs/src/main/asciidoc/qute.adoc]: The Built in template methods work only if you use the `qute` extension in Quarkus, they are not available if you use Qute standalone, that's why I kept it in `qute.adoc`, but I am fine to move it if you think it's better

Comment (mkouba) [reply docs/src/main/asciidoc/qute.adoc]: `qute.adoc` is a simple quickstart guide, `qute-reference.adoc` is a reference guide that contains both the core/standalone and quarkus-specific features.

**Comment by gastaldi** [comment]
@mkouba Changes pushed

**Comment by gsmet** [comment]
@mkouba should we merge that one?

**Review by mkouba — APPROVED** [review]

**Pull request merged**